### ISSUE TITLE
CRA/webpack: Failed to parse source map from typed.ts

### ISSUE THREAD

**Issue by Linus-Boehm** [rayepps/radash#69]
Hey, we are using creat-react-app which uses webpack 5 and getting the following warning when using radash.

```
WARNING in ./node_modules/radash/dist/typed.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/frontend/node_modules/radash/src/typed.ts' file: Error: ENOENT: no such file or directory, open '/frontend/node_modules/radash/src/typed.ts'
 @ ./node_modules/radash/dist/index.js 168:14-32
 @ ./src/modules/buyer-side/contacts/components/ContactImporter/helpers.ts 5:0-34 54:46-54
 @ ./src/modules/buyer-side/contacts/components/ContactImporter/ContactImporter.tsx 12:0-100 65:15-29
 @ ./src/modules/buyer-side/contacts/routes/ContactImportView.tsx 9:0-105 43:35-50
 @ ./src/routes.tsx 289:15-81
 @ ./src/App.tsx 25:0-35 91:49-58
 @ ./src/index.tsx 17:0-26 62:47-50
 ```

Seems like source maps are not emitted. Can someone point me in a direction on how to fix this? Would love to contribute a fix here!
 
**Related:** 
 https://github.com/facebook/create-react-app/pull/11752
 
 Best regards
 Linus

**Comment by ianriosbaf** [comment]
getting the same error here

```
Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\array.ts' file: Error: ENOENT: no 
such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\array.ts'

Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\async.ts' file: Error: ENOENT: no 
such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\async.ts'

Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\curry.ts' file: Error: ENOENT: no 
such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\curry.ts'

Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\index.ts' file: Error: ENOENT: no 
such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\index.ts'

Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\number.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\number.ts'

Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\object.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\object.ts'

Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\random.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\random.ts'

Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\series.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\series.ts'

Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\string.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\string.ts'

Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\typed.ts' file: Error: ENOENT: no 
such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\typed.ts'

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

WARNING in ./node_modules/radash/dist/array.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\array.ts' file: Error: ENOENT: no 
such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\array.ts'

WARNING in ./node_modules/radash/dist/async.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\async.ts' file: Error: ENOENT: no 
such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\async.ts'

WARNING in ./node_modules/radash/dist/curry.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\curry.ts' file: Error: ENOENT: no 
such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\curry.ts'

WARNING in ./node_modules/radash/dist/index.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\index.ts' file: Error: ENOENT: no 
such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\index.ts'

WARNING in ./node_modules/radash/dist/number.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\number.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\number.ts'

WARNING in ./node_modules/radash/dist/object.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\object.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\object.ts'

WARNING in ./node_modules/radash/dist/random.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\random.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\random.ts'

WARNING in ./node_modules/radash/dist/series.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\series.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\series.ts'

WARNING in ./node_modules/radash/dist/string.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\string.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\string.ts'

WARNING in ./node_modules/radash/dist/typed.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\user\repo\node_modules\radash\src\typed.ts' file: Error: ENOENT: no 
such file or directory, open 'C:\Users\user\repo\node_modules\radash\src\typed.ts'
```

**Comment by rayepps** [comment]
Hey all, I'm disappointed this isn't working for CRA. I'll be digging into it this week.

**Comment by Linus-Boehm** [comment]
Awesome @rayepps !
Let me now when I can help you in any way 😊

**Comment by matteosacchetto** [comment]
I believe this should be relatively easy to solve!
Since sourcemaps point to the source files, I think that to solve this issue also the `src` directory should be budled in the npm package

We could filter out tests when packing, and keep only the actual source code together with the compiled one

**Comment by matteosacchetto** [comment]
Nevermind, it seems an unnecessary fix.

@Linus-Boehm, @ianriosbaf could you please install the latest version of radash available at the time of writing (`8.0.3`) and check if this is still an issue.

I've performed the follwing steps:
- create-react-app
- `npm install radash@8.0.3`
- `npm run build`

With version `8.0.3` I do not see issues related to sourcemaps
While with version `7.1.0` (latest previous major version) the issues are there

Please, check if with the latest version the issue is still there and if not I think we can close this issue

**Comment by Linus-Boehm** [comment]
Yes this is fixed with version `8.0.3` 
TY very much 😊

**Comment by ianriosbaf** [comment]
all fixed, thanks!
